.github/workflows/docker-build-push-multiarch.md

```diff
@@ -0,0 +1,98 @@
+# docker-build-push-multiaarch
+
+This is a reusable workflow that uses Grafana's hosted runners to natively build and push multi-architecture docker
+images.
+
+Right now this supports pushing images to:
+
+- Google Artifact Registry
+- DockerHub
+
+And supports building the following image types:
+
+- linux/arm64
+- linux/amd64
+
+## How it works
+
+This generates a matrix based off of the `platforms` input, then creates a job per platform that runs the composite
+actions [docker-build-push-image] and [docker-export-digest] to build and push docker images, and capture their digests.
+There is a then a final job that runs the composite action [docker-import-digests-push-manifest] to push the final
+docker manifest.
+
+[docker/build-push-action]: https://github.com/docker/build-push-action
+[docker-build-push-image]: ../../docker-build-push-image/README.md
+[docker-export-digest]: ../../docker-export-digest/README.md
+[docker-import-digests-push-manifest]: ../../docker-import-digests-push-manifest/README.md
+
+<!-- x-release-please-start-version -->
+
+```yaml
+name: Build and Push and Push MultiArch
+
+on: push
+
+jobs:
+  build-push-multiarch:
+    uses: grafana/shared-workflows/.github/workflows/build-and-push-docker-multiarch.yml@rwhitaker/multi-arch-builds # TODO: Pin to version
```

Comment:
Keep in mind that you cannot really "release" reusable workflows - see [issue here](https://github.com/googleapis/release-please/issues/1366). Users will still be able to use the reusable workflow based on a commit sha, but release-please isn't able to create semver compatible versions. At least not yet!